tests/test_typing/bad/err2.catala_en

```diff
@@ -23,7 +23,7 @@ Error coming from typechecking the following expression:
    |                                  ^^^^^^^^
    + 
 
-Type Structure array coming from expression:
+Type any[10] array coming from expression:
```

Comment:
So this is not an improvement. The union find representative chosen should not be a type variable if there is a non-type-variable representative.